pkg/vm/engine/disttae/txn_table.go

```diff
@@ -2273,3 +2267,16 @@ func (tbl *txnTable) readNewRowid(vec *vector.Vector, row int,
 func (tbl *txnTable) newPkFilter(pkExpr, constExpr *plan.Expr) (*plan.Expr, error) {
 	return plan2.BindFuncExprImplByPlanExpr(tbl.proc.Load().Ctx, "=", []*plan.Expr{pkExpr, constExpr})
 }
+
+func (tbl *txnTable) getSnapshotWriteOffset() int {
+	tbl.Lock()
+	defer tbl.Unlock()
+	return tbl.snapshotWriteOffset
+}
+
+// Before executing an SQL statement, take a snapshot to ensure that the SQL statement does not read the data it writes.
```

Comment:
why do you need this -- we have just got rid of the write offset in table, why you want it back?